src/libNode/PoWProcessing.cpp

```diff
@@ -91,63 +89,131 @@ bool Node::StartPoW(const uint64_t& block_num, uint8_t difficulty,
         vector<unsigned char> mixhash_vec
             = DataConversion::HexStrToUint8Vec(winning_result.mix_hash);
 
-        // Send PoW result
-        // Message = [8-byte block number] [4-byte listening port] [33-byte public key]
-        // [8-byte nonce] [32-byte resulting hash] [32-byte mixhash] [64-byte Signature]
-        vector<unsigned char> powmessage
-            = {MessageType::DIRECTORY, DSInstructionType::POWSUBMISSION};
-        unsigned int cur_offset = MessageOffset::BODY;
+        // Possibles scenario
+        // 1. Found solution that meet ds difficulty and difficulty
+        // - Submit solution
+        // 2. Found solution that met only diffiulty
+        // - Submit solution and continue to do PoW till DS difficulty met or
+        //   ds block received. (stopmining())
+        if (POW::GetInstance().CheckSolnAgainstsTargetedDifficulty(
+                (std::string)winning_result.result, ds_difficulty))
```

Comment:
The winning_result.result is already std::string type, here no need to do a conversion.

Reply:
good poing